vispy/app/canvas.py

```diff
@@ -632,7 +632,9 @@ def __init__(self, type, pos=None, button=None, buttons=None,
         Event.__init__(self, type, **kwargs)
         self._pos = np.array([0, 0]) if (pos is None) else np.array(pos)
         self._button = int(button) if (button is not None) else None
-        self._buttons = [] if (buttons is None) else buttons
+        # Explicitly add button to buttons
+        self._buttons = [] if (buttons is None) else buttons + (
+            [button] if button is not None and type == 'mouse_press' else [])
```

Comment:
I found this bit a little hard to understand. Does splitting it into multiple lines and updating the comment still make sense?

```suggestion
        # Explicitly add button to buttons if newly pressed
        newly_pressed_buttons = [button] if button is not None and type == 'mouse_press' else []
        self._buttons = [] if (buttons is None) else buttons + newly_pressed_buttons
```

Reply:
Would it be fine, if I add the button in a new line. Something like 

```
        self._buttons += [button] if button is not None and type == 'mouse_press' else []

```

Reply:
I will update the comment, the best would be to mention this issue if its allowed.

Reply:
The only problem with doing that `+=` is that it modifies the original `buttons` list inplace, right? We don't want to modify the callers arguments if possible. You could do `self._buttons = self._buttons + ...`. The only argument I have for my way is that the variable name gives some context for what is being done and why (possibly making the comment unnecessary).

If you want to add a comment linking to the issue then that's fine by me.

Reply:
Yes, it would modify the original arguments. Thanks it completely skipped my mind. 

I will make the changes.